NI.package.cmake

```diff
@@ -0,0 +1,195 @@
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/mp3/mp3_custom_source.h"
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/mp4/mp4_file_source.h"
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/mp4/mp4_vector_source.h"
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/mp4/mp4_custom_source.h"
+  )
+endif()
+if ( APPLE )
+  list( APPEND audiostream_sources
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/os/mac/core_audio_source.h"
+  )
+endif()
+if ( WIN32 )
+  list( APPEND audiostream_sources
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/os/win/media_foundation_helper.h"
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/os/win/media_foundation_source.h"
+  )
+endif()
+if ( IOS )
+  list( APPEND audiostream_sources
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/os/ios/avassetreader_source.h"
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/os/ios/avassetreader_source.mm"
+  )
+endif()
+if ( WIN32 AND NIMEDIA_ENABLE_WMA_DECODING )
+  list( APPEND audiostream_sources
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/wma/wma_custom_source.h"
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/wma/wma_file_source.h"
+    "${NI_NIMEDIA_PACKAGE_PATH}/audiostream/src/ni/media/audio/wma/wma_vector_source.h"
+  )
+endif()
+target_sources( audiostream PUBLIC ${audiostream_headers} PRIVATE ${audiostream_sources})
+
+# codec libraries
+find_package( NIFLAC REQUIRED )
+list( APPEND codec_libraries flacpp )
+find_package( NILibVorbis REQUIRED )
+list( APPEND codec_libraries libvorbis )
+if ( APPLE )
+  if ( NOT IOS )
+    find_library( CORESERVICES_FRAMEWORK CoreServices )
+    list( APPEND codec_libraries ${CORESERVICES_FRAMEWORK} )
+  else()
+    foreach( lib IN ITEMS Foundation AVFoundation CoreMedia )
+      find_library( ${lib}_FRAMEWORK ${lib} )
+      if ( NOT ${lib}_FRAMEWORK )
+        message( FATAL_ERROR "could not locate ${lib}.framework" )
+      endif()
+      list( APPEND codec_libraries ${${lib}_FRAMEWORK} )
+    endforeach()
+  endif()
+endif()
+if ( WIN32 )
+  list( APPEND codec_libraries mfplat.lib mfreadwrite.lib mfuuid.lib Propsys.lib )
+endif()
+
+target_link_libraries( audiostream PUBLIC pcm
+                                   PRIVATE boost_iostreams boost_filesystem boost_system ${codec_libraries}
+                     )
+
+target_compile_definitions( audiostream PRIVATE
+  NIMEDIA_ENABLE_AIFF_DECODING=1
+  NIMEDIA_ENABLE_FLAC_DECODING=1
+  NIMEDIA_ENABLE_MP3_DECODING=1
+  NIMEDIA_ENABLE_MP4_DECODING=1
+  NIMEDIA_ENABLE_OGG_DECODING=1
+  NIMEDIA_ENABLE_WAV_DECODING=1
+  NIMEDIA_ENABLE_WMA_DECODING=$<BOOL:${NIMEDIA_ENABLE_WMA_DECODING}> #wma support is off by default, clients need to turn it on
+  NIMEDIA_ENABLE_WAV_ENCODING=1
+  NIMEDIA_ENABLE_ITUNES_DECODING=1
+)
+
+add_library( ni-media INTERFACE )
+target_link_libraries( ni-media INTERFACE audiostream pcm )
```

Comment:
You may consider to add an alias `ni::media`, similar to [here](https://github.com/midi2-dev/ni-midi2/blob/main/NI.package.cmake#L57).

Reply:
Done.